### ISSUE TITLE
Inconsistency between FindMode.Enabled and FindMode.Disabled

### ISSUE THREAD

**Issue by badandbest** [Facepunch/sbox-issues#4401]
### Describe the bug

`FindMode.Enabled` only finds a component if both it and the game object are enabled. Makes perfect sense. Keep that.

But by that logic. `FindMode.Disabled` should find a component if either it or the game object are disabled. But this isn't the case. It only cares if the component itself is disabled or enabled.

### To Reproduce

```csharp
var component = Components.Get<MyComponent>( FindMode.EnabledInSelfAndChildren);
```
 The state of the game object affects the search.

```csharp
var component = Components.Get<MyComponent>( FindMode.DisabledInSelfAndChildren );
```
The state of the game object _does not_ affect the search.

### Expected behavior

If a game object is disabled it should mean its components also count as disabled in a search.

Basically FindMode.Disabled should succeed in every situation that FindMode.Enabled fails.

**Comment by garrynewman** [comment]
Should be fixed in next build
